src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1249,16 +1249,19 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
                     changeVector, lastModifiedTicks);
 
                 var currentRevisionsCount = GetRevisionsCount(context, id);
+
                 if (needToDeleteMore && currentRevisionsCount > 0)
                     moreWork = true;
 
                 if (currentRevisionsCount == 0)
                 {
+                    var res = _documentsStorage.GetDocumentOrTombstone(context, lowerId, throwOnConflict: false);
                     // need to strip the HasRevisions flag from the document
-                    var document = _documentsStorage.Get(context, id);
+                    if (res.Tombstone != null)
+                        _documentsStorage.Delete(context, lowerId, id, null, nonPersistentFlags: NonPersistentDocumentFlags.ByEnforceRevisionConfiguration);
 
-                    if (document != null)
-                        _documentsStorage.Put(context, id, null, document.Data.Clone(context),
+                    if (res.Document != null)
```

Comment:
Can Tombstone and Document be not null at the same time?

Reply:
nope